### PR TITLE
#166840930 User Tag Article 

### PR DESCRIPTION
**Pull request by tohbay**
#### What does this PR do?
Implement User tag articles

#### Description of Task to be completed?
```
- ensure unit tests are written to test the functionality to be implemented
- ensure that data model for tag articles feature was set up with sequelize
- ensure that user can add tags to the articles they author
- ensure there are validations implemented on the feature
- update swagger documentation for the features implemented
```
#### How should this be manually tested?
The following steps can be used:
1) Clone the repository 
2) Cd into the project folder 
3) Run `npm install` to install all project dependencies. 
4) On the terminal run `npm run start:dev`
5) Make the following http calls on Postman:
        `POST  localhost:3000/api/v1/articles`
        `GET  localhost:3000/api/v1/articles`
        `GET  localhost:3000/api/v1/articles/:articleId`
        `PUT  localhost:3000/api/v1/articles/:articleId`
There is a `tag` element in the corresponding response to the above endpoints respectively as shown in the pictures below.

#### Any background context you want to provide?
- Ensure that `node`, `npm` and `postman` are installed on your machine

#### What are the relevant pivotal tracker stories?
Feature 166840930: https://www.pivotaltracker.com/story/show/166840930

#### Screenshots (if appropriate)
Create Article Endpoint
image

Get All Articles Endpoint
image

Update A Specific Article Endpoint
image

### CODE REVIEW

**Comment by codexempire** [comment]
Nice work @tobechuwku please can you kindly rebase the **PR** so that only implementation for the tags show in file changes.

File: src/repository/base.repository.js [hunk 1]
```diff
@@ -72,8 +94,8 @@ class BaseRepository {
    * @returns {object} - returns a database object
    * @memberof BaseRepository
    */
-  static async findAll(model, options) {
```
Comment (kechyy) [inline src/repository/base.repository.js]: Good Job so far @tobechukwuobitube . However, I noticed the baseRepository findAll method was removed, I think you need to confirm no one else used it before it get merged.

File: src/test/article.spec.js [hunk 2]
```diff
@@ -4,10 +4,12 @@ import sinon from 'sinon';
 import app from '../index';
 import BaseRepository from '../repository/base.repository';
 import { createUser, createArticle, generateArticle } from './utils/helpers';
+import { articleSample } from './mockdata/articledata';
 import db from '../database/models';
 import helper from '../helpers/utils';
 
 const ARTICLES_API = '/api/v1/articles';
+const articleEndpoint = '/api/v1/articles';
```
Comment (kechyy) [inline src/test/article.spec.js]: You may want to remove one end point if you're not using the both. Noticed the are the same route.

**Review by kechyy — CHANGES_REQUESTED** [review]
Good Job @tobechukwuobitube . Please make the few changes I spotted.

**Review by kechyy — APPROVED** [review]

**Review by kechyy — APPROVED** [review]
Good Job @tobechukwuobitube and @codexempire . This looks  good to me.

File: src/helpers/tagArticle.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * @param {*} tag - tag
+ * @param {*} articleId - id of article
+ * @returns {object} - - tagged article array
+ */
+const articleTag = async (tag, articleId) => {
+  const tags = tag.split(' ');
+  const addedTags = [];
+  const articlesTagged = [];
+  const createdTags = await Promise.all(
+    tags.map(tagName =>
```
Comment (timi-codes) [inline src/helpers/tagArticle.js:17]: Great implementation @codexempire @tobechukwuobitube

**Review by timi-codes — APPROVED** [review]
This looks good to me. Good Job @codexempire @tobechukwuobitube

**Review by Onyimatics — APPROVED** [review]
Good job @codexempire and @tobechukwuobitube.
 I like your implementation.

**Pull request merged**